### PR TITLE
mysql deployment configurations

### PR DESCRIPTION
**Pull request by HASH10DEV**
please review

### CODE REVIEW

File: deployments/cart/helm/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Comment (kishalayb18) [inline deployments/cart/helm/.helmignore]: remove this file

File: deployments/mysql/helm/template/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mysql
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: parameterized the hardcoded values

File: deployments/mysql/helm/template/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        service: mysql
+    spec:
+      {{ if .Values.psp.enabled }}
+      serviceAccountName: robot-shop
+      {{ end }}
+      containers:
+      - name: mysql
+        image: {{ .Values.image.repo }}/rs-mysql-db:{{ .Values.image.version }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        # added for Istio
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: why is this required?

File: deployments/mysql/helm/template/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        ports:
+        - containerPort: 3306
+        resources:
+          limits:
+            cpu: 200m
+            memory: 1024Mi
+          requests:
+            cpu: 100m
+            memory: 700Mi
+      restartPolicy: Always
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: remove the  affinity,  nodeSelector, tolerations blocks for now

File: deployments/mysql/helm/template/service.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: service
+metadata:
+  labels:
+    service: mysql
+  name: mysql
+spec:
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/service.yaml:7]: provide the type as the NodePort

File: deployments/mysql/helm/template/service.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: service
+metadata:
+  labels:
+    service: mysql
+  name: mysql
+spec:
+  ports:
+  - name: mysql
+    port: 3306
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/service.yaml]: ### {{ .Values.service.ports.port }}

parameterize the values

File: deployments/mysql/helm/values.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+  
+service:
+  type: ClusterIP
```
Comment (kishalayb18) [inline deployments/mysql/helm/values.yaml]: modify the type to NodePort for now

File: deployments/mysql/helm/values.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
```
Comment (kishalayb18) [inline deployments/mysql/helm/values.yaml]: remove the values which is not required

**Review by kishalayb18 — CHANGES_REQUESTED** [review]
resolve the changes

File: deployments/mysql/helm/values.yaml (hunk 7 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/values.yaml]: this is done

File: deployments/mysql/helm/template/deployment.yaml (hunk 4 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/template/deployment.yaml]: this is done

File: deployments/mysql/helm/values.yaml (hunk 8 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/values.yaml]: done

File: deployments/mysql/helm/template/deployment.yaml (hunk 2 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/template/deployment.yaml]: done

File: deployments/mysql/helm/template/deployment.yaml (hunk 3 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/template/deployment.yaml]: removed

File: deployments/mysql/helm/template/service.yaml (hunk 6 above)
Comment (HASH10DEV) [reply deployments/mysql/helm/template/service.yaml]: done

**Comment by HASH10DEV** [comment]
hi kishalay ive made all the changes accordingly

File: deployments/mysql/helm/template/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .values.appname }}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: please resolve this
{{ .Values.appName }}

File: deployments/mysql/helm/template/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+kind: Deployment
+metadata:
+  name: {{ .values.appname }}
+  labels:
+    service: {{ .values.appname }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      service: {{ .values.appname }}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: modify this to {{ .Values.appName }}

File: deployments/mysql/helm/template/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  template:
+    metadata:
+      labels:
+        service: {{ .values.appname }}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: {{ .Values.appName }}

File: deployments/mysql/helm/template/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    spec:
+      {{ if .Values.psp.enabled }}
+      serviceAccountName: {{ .Values.psp.serviceAccountName }}
+      {{ end }}
+      containers:
+      - name: {{ .values.appname }}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml:21]: {{ .Values.appName }}

File: deployments/mysql/helm/template/deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        image: {{ .Values.containers.image.repo }}{{ .Values.containers.image.tag }}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml]: there must be a colon
{{ .Values.containers.image.repo }}:{{ .Values.containers.image.tag }}

File: deployments/mysql/helm/template/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        imagePullPolicy: {{ .Values.containers.image.pullPolicy }}
+        securityContext:
+          capabilities:
+            add: ["NET_ADMIN"]
+        ports:
+        - containerPort: {{ .Values.service.ports.targetPort }}
+        {{- include "container.resources" . | nindent 8 -}}
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/deployment.yaml:29]: has the resource been added to helpers file, please verify

**Review by kishalayb18 — CHANGES_REQUESTED** [review]
modify the files based on the feedback please

File: deployments/mysql/helm/template/ingress.yaml [hunk 15]
```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.k8s.io/v1
```
Comment (kishalayb18) [inline deployments/mysql/helm/template/ingress.yaml]: this file is not required

File: deployments/mysql/helm/values.yaml [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+
+  runAs:
+    user: 100
+    group: {}
+
+service:
+  type: NodePort
+  ports:
+    name: http
+    port: 8080
```
Comment (kishalayb18) [inline deployments/mysql/helm/values.yaml]: compare this with main repository

File: deployments/mysql/helm/values.yaml [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+  enabled: false
+  serviceAccountName: robot-shop
+
+containers:
+  image:
+    repo: robotshop/rs-mysql
+    tag: latest
+    pullPolicy: ifNotPresent
+
+  resources:
```
Comment (kishalayb18) [inline deployments/mysql/helm/values.yaml:14]: compare this with main repository

**Review by kishalayb18 — APPROVED** [review]
LGTM

**Comment by kishalayb18** [comment]
good to ship

**Pull request merged**